app/ui-cached-collection/client/models/CachedCollection.js

```diff
@@ -27,31 +45,25 @@ class CachedCollectionManagerClass {
 
 		// Wait 1s to start or the code will run before the connection and
 		// on first connection the `reconnect` callbacks will run
-		Meteor.setTimeout(() => {
-			let connectionWasOnline = true;
-			Tracker.autorun(() => {
-				const { connected } = Meteor.connection.status();
-
-				if (connected === true && connectionWasOnline === false) {
-					for (const cb of this.reconnectCb) {
-						cb();
-					}
-				}
-
-				connectionWasOnline = connected;
-			});
-		}, 1000);
-
 		Tracker.autorun(() => {
-			if (Meteor.userId() !== null) {
-				if (this.logged === false) {
-					for (const cb of this.loginCb) {
-						cb();
-					}
-				}
+			this.step = this.step || 0;
+			const { connected } = Meteor.status();
+			switch (this.step) {
+				case 0:
+					return !connected || this.step++;
+				case 1:
+					return connected || this.step++;
+				case 2:
```

Comment:
Can we have some nice constants for those magic numbers?

Reply:
they are not so magic, they have the same scope, but if you argue all the block has no sense without some tip, maybe I should agree with you